lib/util/http2_rpc.c

```diff
@@ -44,6 +45,89 @@ typedef struct http2_rpc_data {
 	size_t data_length;
 } http2_rpc_data;
 
+static int decompress_grpc_response(http2_rpc_handle *hd)
+{
+	if (hd->bytes_written < 7) {
+		return 1;
+	}
+
+	// Check if gRPC response starts with compression flag and has gzip magic bytes
+	if ((unsigned char)hd->output_buffer[0] != 0x01) {
+		// No compression flag found, data appears uncompressed
+		return 1;
+	}
+
+	// Check for gzip magic bytes
+	if ((unsigned char)hd->output_buffer[5] != 0x1f || (unsigned char)hd->output_buffer[6] != 0x8b) {
+		// No gzip magic bytes found at offset
+		return -1;
+	}
+
+	z_stream stream;
```

Comment:
Instead of using `memset` you could use:
```suggestion
	z_stream stream = { 0 };
```